dips/tests/test_views.py

```diff
@@ -0,0 +1,42 @@
+from django.test import TestCase
+from django.test.client import RequestFactory
+from unittest.mock import ANY, patch
+
+from dips.models import User
+from dips.views import search
+
+
+class ViewsTests(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_superuser(
+            'admin', 'admin@example.com', 'admin')
+        self.factory = RequestFactory()
+
+    @patch('elasticsearch_dsl.Search.execute')
+    @patch('elasticsearch_dsl.Search.count', return_value=0)
+    @patch('dips.views.messages.error')
+    @patch('dips.views.render')
+    def test_search_wrong_dates(self, mock, mock_2, patch, patch_2):
+        request = self.factory.get(
+            '/search', {'start_date': '2018/01/01', 'end_date': 'Nov. 6, 2018'})
+        request.user = self.user
+        search(request)
+        expected_filters = {
+            'formats': [],
+            'collections': [],
+            'start_date': '2018/01/01',
+            'end_date': 'Nov. 6, 2018',
+        }
+        # Wrong formats should be maintained in filters
+        mock.assert_called_with(request, 'search.html', {
```

Comment:
This seems pretty specific. If I add something unrelated to the request context, it will break this test, which isn't ideal. Mock is kind of limited here, but you can do something like `patch_2.call_args[2]['filters']` (untested, may need tweaking) to just check the filters.

I think maybe a more robust approach though is to not mock render, and then to test the response object. You can [just check the response.context](https://docs.djangoproject.com/en/2.2/topics/testing/tools/#testing-responses) to make sure the filters are set properly.